src/components/molecules/Visualizer/Engine/Cesium/cameraLimiter.ts

```diff
@@ -0,0 +1,237 @@
+import {
+  Cartesian3,
+  Cartographic,
+  Color,
+  EllipsoidGeodesic,
+  PolylineDashMaterialProperty,
+  Math,
+  Camera as CesiumCamera,
+  Rectangle,
+} from "cesium";
+import type { Viewer as CesiumViewer } from "cesium";
+import { useEffect, useMemo, useState } from "react";
+import { CesiumComponentRef } from "resium";
+import { RefObject } from "use-callback-ref/dist/es5/types";
+
+import { Camera } from "@reearth/util/value";
+
+import type { SceneProperty } from "..";
+
+import { getCamera } from "./common";
+
+export function useCameraLimiter(
+  cesium: RefObject<CesiumComponentRef<CesiumViewer>>,
+  camera: Camera | undefined,
+  property: SceneProperty["cameraLimiter"] | undefined,
+) {
+  const geodsic = useMemo(():
+    | undefined
+    | { geodesicVertical: EllipsoidGeodesic; geodesicHorizontal: EllipsoidGeodesic } => {
+    const viewer = cesium.current?.cesiumElement;
+    if (!viewer || viewer.isDestroyed()) return undefined;
+    return getGeodsic(viewer, property);
+  }, [cesium, property]);
+
+  // calculate inner limiter dimensions
+  const targetWidth = 1000000;
+  const targetLength = 1000000;
```

Comment:
Probably something we need to discuss a bit, but I really feel like these variables and the corresponding user-facing properties should be renamed since target really isn't meaningful. With that change, I feel like we can get rid of the explanatory comment as well (I don't feel like this place is complex or generic enough to need comments. Just a problem with variable naming).
Maybe we can discuss a bit before merging if we have time.

Reply:
I think this can be done in the future release of Camera limiter (and I'll open an issue for it as well), as there are some UI changes as well in future.
For now the focus will be on code style.